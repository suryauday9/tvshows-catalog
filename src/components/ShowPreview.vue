<template>
  <div v-if="show">
      <v-card
        class="ma-1"
        height="180"
        width="180"
        @click="viewDetails(show.id)"
      >
        <v-img
          v-if="show.image != null"
          :src="show.image.medium"
          height="180"
          class="white--text align-end"
        >
          <v-card-title class="subtitle-2 black pa-1">
            <div class="d-flex align-center show-name">
              {{ show.name }}
            </div>
            <div class="d-flex align-center show-rating" v-if="show.rating && show.rating.average">
              {{ show.rating.average }}
              <v-icon color="red">mdi-star</v-icon>
            </div>
          </v-card-title>
        </v-img>
        <v-img
          v-else
          :src="require('../assets/tv-show-img.jpg')"
          height="180"
          class="white--text align-end"
        >
          <v-card-title class="subtitle-2 black pa-3">
            {{ show.name }}
          </v-card-title>
        </v-img>
      </v-card>
  </div>
</template>

<script>
export default {
  name: "ShowPreview",
  props: {
    show: {}
  },
  methods: {
    viewDetails(id) {
      this.$router.push({ name: "show", params: { id: id } });
    }
  }
};
</script>
<style scoped>
.show-name {
  width: 100%;
  display: block !important;
}
.show-rating {
  width: 100%;
  display: block !important;
}
</style>
