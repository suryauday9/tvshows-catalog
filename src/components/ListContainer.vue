<template>
  <div>
    <v-sheet class="mx-auto white darken-4">
      <div class="headline font-weight-bold blue-grey--text mx-12 mt-12">
        {{ genre }}
      </div>
      <v-slide-group class="pa-4" show-arrows="desktop" light>
        <v-slide-item v-for="show in showsByGenre(shows, genre)" :key="show.id">
          <show-preview :show="show" />
        </v-slide-item>
      </v-slide-group>
    </v-sheet>
  </div>
</template>

<script>
import ShowPreview from "../components/ShowPreview.vue";
export default {
  name: "ListContainer",
  components: {
    ShowPreview
  },
  props: {
    genre: String
  },
  methods: {
    showsByGenre: function (shows, genre) {
      return shows
        .filter((show) => show.genres && show.genres.some((g) => g === genre))
        .slice(0, 19)
    }
  },
  computed: {
    shows() {
        return this.$store.state.shows;
    }
  }
};
</script>
<style scoped>
.headline {
  text-align: left;
  text-decoration: underline;
}
</style>