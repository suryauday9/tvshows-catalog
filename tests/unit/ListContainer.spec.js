import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import { initializeApp } from './initializeApp';
import ListContainer from '../../src/components/ListContainer.vue';
import { showsData } from './response';
import ShowPreview from '../../src/components/ShowPreview.vue';

const vuetify = new Vuetify();
const localVue = initializeApp(createLocalVue());

describe('ListContainer component', () => {  
    let wrapper;
    
    const store = new Vuex.Store({
        state: {
            shows: showsData,
            genres: [],
            showDetails: [],
            showThumbnails: [],
            showSearchbar: false
        },
        actions: {}
    });

    const mountComponent = (options) => {
        return shallowMount(ListContainer, {
            localVue,
            store,
            // computed: {
            //     shows() {
            //         return store.state.shows;
            //     }
            // },
            vuetify,
            ...options,
        })
    }

    beforeEach(() => {
        wrapper = mountComponent();
    })

    afterEach(() => {
        wrapper.destroy();
    })

    it('renders component correctly', () => {
        // see if component renders
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should be a Vue instance', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    it('should get 3 results for Action genre', () => {
        const results = wrapper.vm.showsByGenre(store.state.shows, 'Action');
        expect(results.length).toEqual(3);
    });

    it('rendered component data should match with store', () => {
        // see if data loads
        expect(wrapper.vm.shows).toBe(store.state.shows);
    });

    it('it should load ShowPreview component', () => {
        expect(ShowPreview).toBeTruthy();
    });

});