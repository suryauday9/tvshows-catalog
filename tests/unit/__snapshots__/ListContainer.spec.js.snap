// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer component renders component correctly 1`] = `
<div>
  <v-sheet-stub
    class="mx-auto white darken-4"
    tag="div"
  >
    <div
      class="headline font-weight-bold blue-grey--text mx-12 mt-12"
    >
      
      
    
    </div>
     
    <v-slide-group-stub
      activeclass="v-slide-item--active"
      class="pa-4"
      light="true"
      mobilebreakpoint="1264"
      nexticon="$next"
      previcon="$prev"
      showarrows="desktop"
      tag="div"
    />
  </v-sheet-stub>
</div>
`;
